src/common/timer.c

```diff
@@ -86,8 +86,18 @@ void timesdiv(struct timespec *tvs, COUNTER div)
 }
 
 void
-init_timestamp(timestamp_t *ctx)
+init_timestamp(struct timespec *timestamp)
 {
-    timerclear(ctx);
+    timesclear(timestamp);
 }
 
+int get_current_time(struct timespec *ts){
+    #if defined _POSIX_C_SOURCE  && _POSIX_C_SOURCE >= 199309L
+        return clock_gettime(CLOCK_REALTIME, ts);
+    #else
+        struct timeval tv;
+        int success = gettimeofday(&tv, NULL);
+        TIMEVAL_TO_TIMESPEC(&tv, ts);
+        return success;
```

Comment:
nice that this supports older OS's. How much testing has there been on this?

Reply:
A comprehensive test on this is definitely missing, I made some user tests with basic setups. However there are many include guards, with if, elseif etc... branches, they all need to be tested separately. Just to mention one, it wasn't tested how it works with netmap